src/ja/worker/config.py

```diff
@@ -1,36 +1,76 @@
 from typing import Dict
-
 from ja.common.config import Config
 from ja.common.proxy.ssh import SSHConfig
 from ja.common.work_machine import ResourceAllocation
+import yaml
+import os
 
 
 class WorkerConfig(Config):
     """
     Config for the worker client.
     """
 
+    def __init__(self, conf_path: str):
+        """!
+        reads the config file from the disk
+        creates resource allocation instance
+        """
+        self._conf_path = conf_path
+        self._ssh_config = self.read_conf()
+        self._resource_allocation = self.get_resources()
+
+    def read_conf(self) -> SSHConfig:
+        with open(self._conf_path, 'r') as stream:
+            data_loaded = yaml.safe_load(stream)
+
+        hostname: str = data_loaded["hostname"]
+        username: str = data_loaded["username"]
+        self._uid = data_loaded["uid"]
+        return SSHConfig(hostname, username)
+
+    def get_resources(self) -> ResourceAllocation:
```

Comment:
We do not need this method.
The amount of resources should be read from the configuration file instead of being determined automatically.
This is because the administrator might want to reserve some amount of resources for other processes.

Reply:
Maybe we can have this as a fallback? If nothing is specified in the config file.

Reply:
I think we should not provide a fallback at all.
I think the administrator should be forced to make a decision regarding how many resources the worker client should be using because we cannot provide a default that would work well in all situations.